### PR TITLE
Cloud - Fix toc link

### PR DESCRIPTION
**Pull request by brayunm**

### CODE REVIEW

File: release-notes/Cloud/DCP_change_log.md [hunk 1]
```diff
@@ -9,6 +9,10 @@ The dataminer.services platform gets updated continuously. This change log can h
 > [!NOTE]
 > Many features on dataminer.services are dependent on DxMs. You can find the change logs for these under [DxM release notes](xref:DxM_RNs_index).
 
+#### 10 June 2024 - Fix - ToC checkbox - Fixed an issue with the ToC URL
```
Comment (MariekeGO) [inline release-notes/Cloud/DCP_change_log.md:12]: Is there an intranet record for this RN? Also, could you rephrase the RN so it's more clear for the user what exactly the issue was?

Comment (brayunm) [reply release-notes/Cloud/DCP_change_log.md:12]: I forgot to add the RN to this.
The trailing slash was causing the issue, we temporarily removed it before we'll change it to an aka.ms rule.

File: release-notes/Cloud/DCP_change_log.md [hunk 2]
```diff
@@ -9,6 +9,10 @@ The dataminer.services platform gets updated continuously. This change log can h
 > [!NOTE]
 > Many features on dataminer.services are dependent on DxMs. You can find the change logs for these under [DxM release notes](xref:DxM_RNs_index).
 
+#### 10 June 2024 - Fix - ToC checkbox - Fixed an issue with the ToC URL [ID_39895]
+
+The link of our ToC was updated to the correct link.
```
Comment (MariekeGO) [inline release-notes/Cloud/DCP_change_log.md:14]: Could you rephrase this so it's clear what the issue was? As a user, I wouldn't have a clue what is meant with the current sentence.

Comment (brayunm) [reply release-notes/Cloud/DCP_change_log.md]: Discussed in chat, updated it with the given suggestion

**Review by MariekeGO — APPROVED** [review]
